### DISCUSSION TITLE
Using renovate to update TYPO3 installations minor/patches only

### DISCUSSION THREAD

**Discussion by support-beech** [renovatebot/renovate#17939] [Request Help] [answered]
https://stackoverflow.com/questions/73826812/using-renovate-to-update-typo3-installations-minor-patches-only

Mirror of stackoverflow since renovate stackoverflow didn't seem that active.

I have been experimenting with Renovate and would like to use it to auto update TYPO3 to the latest version. v10.4.x - where Renovate should only create MR's whenever the is an update available for the 'X' part.

I am using the following configuration:

**config.js**
```
    module.exports = {
        endpoint: 'https://gitlab.companyname/api/v4/',
        token: '**gitlab_token**',
        platform: 'gitlab',
        baseBranches: ["master"],
        assignees: ["Me"],
        enabledManagers: "composer",
        onboardingConfig: {
            extends: ['config:base'],
        },
        repositories: [
            {
                repository: 'CMS/sitetemplate',
            },
        ],
    };
```
**renovate.json in the project CMS/sitetemplate**
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    ":disableDependencyDashboard",
    ":separatePatchReleases"
  ],
  "prHourlyLimit": 0,
  "prConcurrentLimit": 0,
  "baseBranches": [
    "master"
  ],
  "enabledManagers": [
    "composer"
  ],
  "commitMessagePrefix": "[TASK] ",
  "commitMessageTopic": "{{depName}}",
  "commitMessageExtra": " ({{{displayFrom}}} => {{{displayTo}}})",
  "rangeStrategy": "update-lockfile",
  "packageRules": [
    {
      "matchUpdateTypes": [
        "patch"
      ],
      "matchPackagePrefixes": [
        "typo3/cms-"
      ],
      "excludePackageNames": [
        "typo3/cms-cli",
        "typo3/cms-composer-installers"
      ],
      "groupName": "TYPO3 CMS",
      "labels": [
        "typo3"
      ]
    },
    {
      "matchUpdateTypes": [
        "minor",
        "pin",
        "digest",
        "major"
      ],
      "enabled": false
    }
  ]
}
```

**.gitlab-ci.yaml**
```
Update Web Project:
  rules:
    - if: $RENOVATE_APPLICATION_TOKEN
      when: never
    - if: $RENOVATE_WEB_TOKEN
      when: manual
  variables:
    RENOVATE_TOKEN: $RENOVATE_WEB_TOKEN
  script:
    - renovate $RENOVATE_EXTRA_FLAGS
  when: manual
```

- I have played around with the config a bit and it will create MR's, issues etc when i have other settings. So its working. However with the current configuration it does not create a PR.
- The sitetemplate currently has TYPO3 V10.4.25 (in lock file) and i want it to create a PR so it will update all typo3/cms-* packages to the lastest TYPO3 V10.4.32
- It should only update the lock file.
- Currently when i run the CI i get no MR's and the following output

**The CI output**
[PASTEBIN since it was to large](https://pastebin.com/raw/iMaNR4Cq)

Anyone has any expirience with this, what configuration am i doing wrong?

**Comment by rarkins** [comment, marked as answer]
At a quick glance I think you're missing a setting of `separateMinorPatch=true`. You could turn that on globally (if you want to separate patch updates from minor for all packages) or specifically for certain packages by using packageRules.

**Reply by support-beech** [reply]
@rarkins Thanks, confirmed working.
